plugins/woocommerce/src/Database/Migrations/CustomOrderTable/PostsToOrdersMigrationController.php

```diff
@@ -71,9 +72,101 @@ public function get_migrated_meta_keys() {
 	public function migrate_orders( array $order_post_ids ): void {
 		$this->error_logger = WC()->call_function( 'wc_get_logger' );
 
+		$using_transactions = $this->maybe_start_transaction();
+		if ( null === $using_transactions ) {
+			return;
+		}
+
+		$errors_were_logged = false;
+
 		foreach ( $this->all_migrators as $migrator ) {
-			$this->do_orders_migration_step( $migrator, $order_post_ids );
+			$errors_were_logged = $this->do_orders_migration_step( $migrator, $order_post_ids );
+			if ( $errors_were_logged && $using_transactions ) {
+				$this->rollback_transaction();
+				break;
+			}
+		}
+
+		if ( ! $errors_were_logged && $using_transactions ) {
+			$this->commit_transaction();
```

Comment:
Do we need to take any action (whether re-running/scheduling a re-run of the migration step, or simply logging the failure) in the event that the commit fails?